pkg/goformatters/analyzer.go

```diff
@@ -0,0 +1,55 @@
+package goformatters
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rogpeppe/go-internal/diff"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/goformatters/internal"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+)
+
+// NewAnalyzer converts a [Formatter] to an [analysis.Analyzer].
+func NewAnalyzer(logger logutils.Log, doc string, formatter Formatter) *analysis.Analyzer {
+	return &analysis.Analyzer{
+		Name: formatter.Name(),
+		Doc:  doc,
+		Run: func(pass *analysis.Pass) (any, error) {
+			for _, file := range pass.Files {
+				position, isGoFiles := goanalysis.GetGoFilePosition(pass, file)
```

Comment:
Perhaps should be without `s`:
```suggestion
				position, isGoFile := goanalysis.GetGoFilePosition(pass, file)
```